app/src/App.vue

```diff
@@ -1,11 +1,40 @@
 <template>
-  <v-app class="torus-app">
+  <v-app class="torus-app" v-if="isOnline">
     <router-view />
   </v-app>
+  <v-app class="torus-app" v-else>
+    <offline />
+  </v-app>
 </template>
 
 <script>
-export default {}
+import { mapState } from 'vuex'
+import Offline from './views/Offline'
+export default {
+  components: {
+    Offline
+  },
+  computed: mapState(['isOnline']),
+  mounted() {
+    if (typeof window !== 'undefined') {
+      const onlineHandler = () => {
+        this.$store.dispatch('updateNetworkState', { infuraNetworkStatus: true })
+      }
+
+      const offlineHandler = () => {
+        this.$store.dispatch('updateNetworkState', { infuraNetworkStatus: false })
+      }
+
+      window.addEventListener('online', onlineHandler)

```

Comment:
Can you move these to infura controller as well?

Reply:
Yes sure. But the `beforeDestroy hook` has to be within the Vue component to listen for changes on the component and trigger the `InfuraController` to act.